fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/ISchemaAdapter.java

```diff
@@ -0,0 +1,622 @@
+     *
+     * @param schemaName
+     * @param procedureName
+     * @param supplier
+     */
+    public void createOrReplaceProcedure(String schemaName, String procedureName, Supplier<String> supplier);
+
+    /**
+     * Drop the given procedure
+     *
+     * @param schemaName
+     * @param procedureName
+     */
+    public void dropProcedure(String schemaName, String procedureName);
+
+    /**
+     * Create a unique index
+     * @param schemaName
+     * @param tableName
+     * @param indexName
+     * @param tenantColumnName
+     * @param indexColumns
+     * @param includeColumns
+     * @param distributionType
+     * @param distributionColumnName
+     */
+    public void createUniqueIndex(String schemaName, String tableName, String indexName, String tenantColumnName,
+            List<OrderedColumnDef> indexColumns, List<String> includeColumns, 
+            DistributionType distributionType, String distributionColumnName);
+
+    /**
+     * Create a unique index
+     * @param schemaName
+     * @param tableName
+     * @param indexName
+     * @param tenantColumnName
+     * @param indexColumns
+     * @param distributionRules
+     * @param distributionColumnName
+     */
+    public void createUniqueIndex(String schemaName, String tableName, String indexName, String tenantColumnName,
+            List<OrderedColumnDef> indexColumns, DistributionType distributionType, String distributionColumnName);
+
+    /**
+     * Create an index on the named schema.table object
+     * @param schemaName
+     * @param tableName
+     * @param indexName
+     * @param tenantColumnName
+     * @param indexColumns
+     * @param distributionType
+     */
+    public void createIndex(String schemaName, String tableName, String indexName, String tenantColumnName,
+            List<OrderedColumnDef> indexColumns, DistributionType distributionType);
+
+    /**
+     *
+     * <pre>
+     * CREATE VARIABLE ptng.session_tenant INT DEFAULT NULL;
+     * </pre>
+     *
+     * @param schemaName
+     * @param variableName
+     */
+    public void createIntVariable(String schemaName, String variableName);
+
+    /**
+     *
+     * <pre>
+     * CREATE OR REPLACE PERMISSION ROW_ACCESS ON ptng.patients FOR ROWS WHERE patients.mt_id =
+     * ptng.session_tenant ENFORCED FOR ALL ACCESS ENABLE;
+     * </pre>
+     *
+     * @param schemaName
+     * @param permissionName
+     * @param tableName
+     * @param predicate
+     */
+    public void createOrReplacePermission(String schemaName, String permissionName, String tableName, String predicate);
+
+    /**
+     *
+     *
+     * <pre> ALTER TABLE <tbl> ACTIVATE ROW ACCESS CONTROL
+     * </pre>
+     *
+     * @param schemaName
+     * @param tableName
+     */
+    public void activateRowAccessControl(String schemaName, String tableName);
+
+    /**
+     * Deactivate row access control on a table ALTER TABLE <tbl> DEACTIVATE ROW
```

Comment:
pretty sure javadoc parse will barf on this

Reply:
old comment (from the database adapter), but I changed it anyway.